pyquil/gate_matrices.py

```diff
@@ -0,0 +1,292 @@
+
+    Z - Pauli-Z :math:`\begin{pmatrix} 1 & 0 \\ 0 & -1 \end{pmatrix}`
+
+    H - Hadamard
+    :math:`\frac{1}{\sqrt{2}} \begin{pmatrix} 1 & 1 \\ 1 & -1 \end{pmatrix}`
+
+    S - PHASE(pi/2)
+    :math:`\begin{pmatrix} 1 & 0 \\ 0 & i \end{pmatrix}`
+
+    T - PHASE(pi/4)
+    :math:`\begin{pmatrix} 1 & 0 \\ 0 & e^{i \pi / 4} \end{pmatrix}`
+
+    PHASE(:math:`\phi`) - PHASE
+    :math:`\begin{pmatrix} 1 & 0 \\ 0 & e^{i \phi} \end{pmatrix}`
+
+    RX(:math:`\phi`) - RX
+    :math:`\begin{pmatrix} \cos(\phi / 2) & -i \sin(\phi/2) \\
+                           -i \sin(\phi/2) & \cos(\phi/2) \end{pmatrix}`
+
+    RY(:math:`\phi`) - RY
+    :math:`\begin{pmatrix} \cos(\phi / 2) & -\sin(\phi / 2) \\
+                           \sin(\phi/2) & \cos(\phi/2) \end{pmatrix}`
+
+    RZ(:math:`\phi`) - RZ
+    :math:`\begin{pmatrix} \cos(\phi/2) - i \sin(\phi/2) & 0 \\
+                           0 & \cos(\phi/2) + i \sin(\phi/2) \end{pmatrix}`
+
+    CZ - controlled-Z
+    :math:`P_0 \otimes I + P_1 \otimes Z = \begin{pmatrix} 1&0&0&0 \\ 0&1&0&0 \\
+                                            0&0&1&0 \\ 0&0&0&-1 \end{pmatrix}`
+
+    CNOT - controlled-X / controlled-NOT
+    :math:`P_0 \otimes I + P_1 \otimes X = \begin{pmatrix} 1&0&0&0 \\ 0&1&0&0 \\
+                                            0&0&0&1 \\ 0&0&1&0 \end{pmatrix}`
+
+    CCNOT - double-controlled-X
+    :math:`P_0 \otimes P_0 \otimes I + P_0 \otimes P_1 \otimes I + P_1 \otimes P_0 \otimes I
+                                     + P_1 \otimes P_1 \otimes X`
+
+    CPHASE00(:math:`\phi`) - controlled-phase-on-|00>
+    :math:`\text{diag}(e^{i \phi}, 1, 1, 1,)`
+
+    CPHASE01(:math:`\phi`) - controlled-phase-on-|01>
+    :math:`\text{diag}(1, e^{i \phi}, 1, 1,)`
+
+    CPHASE10(:math:`\phi`) - controlled-phase-on-|10>
+    :math:`\text{diag}(1, 1, e^{i \phi}, 1)`
+
+    CPHASE(:math:`\phi`) - controlled-phase-on-|11>
+    :math:`\text{diag}(1, 1, 1, e^{i \phi})`
+
+    SWAP - swap
+    :math:`\begin{pmatrix} 1&0&0&0 \\ 0&0&1&0 \\ 0&1&0&0 \\ 0&0&0&1 \end{pmatrix}`
+
+    CSWAP - controlled-swap
+    :math:`P_0 \otimes I_2 + P_1 \otimes \text{SWAP}`
+
+    ISWAP - i-phase-swap
+    :math:`\begin{pmatrix} 1&0&0&0 \\ 0&0&i&0 \\ 0&i&0&0 \\ 0&0&0&1 \end{pmatrix}`
+
+    PSWAP(:math:`\phi`) - phi-phase-swap
+    :math:`\begin{pmatrix} 1&0&0&0 \\ 0&0&e^{i\phi}&0 \\ 0&e^{i\phi}&0&0 \\ 0&0&0&1 \end{pmatrix}`
+
+Specialized gates / internal utility gates:
+    BARENCO(:math:`\alpha, \phi, \theta`) - Barenco gate
+    :math:`\begin{pmatrix} 1&0&0&0 \\ 0&1&0&0 \\ 0&0&e^{i\phi} \cos\theta & -i e^{i(\alpha-\phi)}
+     \sin\theta \\ 0&0&-i e^{i(\alpha+\phi)} \sin\theta & e^{i\alpha} \cos\theta \end{pmatrix}`
+
+    P0 - project-onto-zero
+    :math:`\begin{pmatrix} 1 & 0 \\ 0 & 0 \end{pmatrix}`
+
+    P1 - project-onto-one
+    :math:`\begin{pmatrix} 0 & 0 \\ 0 & 1 \end{pmatrix}`
+"""
+import numpy as np
+import cmath
+
+I = np.array([[1.0, 0.0], [0.0, 1.0]])
```

Comment:
Should these be refactored to be attributes on the existing `gate.py` objects? This would reduce any namespace confusion.